examples/box_game/box_game_spectator.rs

```diff
@@ -51,7 +51,10 @@ fn main() -> Result<(), Box<dyn std::error::Error>> {
         .add_plugins(DefaultPlugins)
         .add_systems(Startup, setup_system)
         // these systems will be executed as part of the advance frame update
-        .add_systems(GgrsSchedule, (move_cube_system, increase_frame_system))
+        .add_systems(
+            GgrsSchedule,
+            (move_cube_system, increase_frame_system).in_set(AdvanceWorldSet::Main),
```

Comment:
I'm not sure I like the extra boiler-plate this `AdvanceWorldSet` adds. Anything that involves time will have to explicitly belong to a set... Maybe we should do it in a separate schedule, although it leads to slightly less parallel code?

Reply:
Agreed, that was a DX regression. I've added a new `AdvanceWorld` schedule (which better aligns with `SaveWorld` and `LoadWorld` naming). Now, the `run_ggrs_schedules` system will run `LoadWorld`, `AdvanceWorld`, and `SaveWorld`. The original `GgrsSchedule` is the run during the `AdvanceWorldSet::Main` set by a dumb system:

```rust
|world: &mut World| world.run_schedule(GgrsSchedule)
```

This allows for easily adding systems before and after the `GgrsSchedule` using `AdvanceWorld` and `AdvanceWorldSet`, and also allows for game logic to just be added to `GgrsSchedule` as before.